cdap-ui/app/hydrator/services/create/stores/preview-store.js

```diff
@@ -41,6 +42,11 @@ var preview = (state = getInitialState(), action = {}) => {
       const {previewId} = action.payload;
       stateCopy.previewId = previewId;
       return Object.assign({}, state, stateCopy);
+    case previewActions.SET_RUNTIME_ARGUMENTS:
+      stateCopy = Object.assign({}, state);
+      const {runtimeArguments} = action.payload;
+      stateCopy.runtimeArguments = runtimeArguments;
+      return Object.assign({}, state, stateCopy);
```

Comment:
We could as well have this as `stateCopy.runTImeArgumenst = action.payload.runtimeArguments`. Not sure if we are gaining anything object dereferencing here.